argoverse/utils/sim2.py

```diff
@@ -0,0 +1,106 @@
+"""
+Utility for 2d rigid body transformations with scaling.
+
+Refs:
+    http://ethaneade.com/lie_groups.pdf
+    https://github.com/borglab/gtsam/blob/develop/gtsam_unstable/geometry/Similarity3.h
+"""
+
+from typing import Union
+
+import numpy as np
+
+from argoverse.utils.helpers import assert_np_array_shape
+
+
+class Sim2:
+    """ Implements the Similarity(2) class."""
+
+    def __init__(self, R: np.ndarray, t: np.ndarray, s: Union[int, float]) -> None:
+        """Initialize from rotation R, translation t, and scale s.
+
+        Args:
+            R: array of shape (2x2) representing 2d rotation matrix
+            t: array of shape (2,) representing 2d translation
+            s: scaling factor
+        """
+        assert_np_array_shape(R, (2, 2))
+        assert_np_array_shape(t, (2,))
+        assert isinstance(s, float) or isinstance(s, int)
+        self.R_ = R.astype(np.float32)
+        self.t_ = t.astype(np.float32)
+        self.s_ = float(s)
+
+    def __eq__(self, other: object) -> bool:
+        """Check for equality with other Sim(2) object"""
+        if not isinstance(other, Sim2):
+            return False
+
+        if not np.isclose(self.scale, other.scale):
+            return False
+
+        if not np.allclose(self.rotation, other.rotation):
+            return False
+
+        if not np.allclose(self.translation, other.translation):
+            return False
+
+        return True
+
+    @property
+    def rotation(self) -> np.ndarray:
+        """Return the 2x2 rotation matrix"""
+        return self.R_
+
+    @property
+    def translation(self) -> np.ndarray:
+        """Return the (2,) translation vector"""
+        return self.t_
+
+    @property
+    def scale(self) -> float:
+        """Return the scale."""
+        return self.s_
+
+    @property
+    def matrix(self) -> np.ndarray:
+        """Calculate 3*3 matrix group equivalent"""
+        T = np.zeros((3, 3))
+        T[:2, :2] = self.R_
+        T[:2, 2] = self.t_
+        T[2, 2] = 1 / self.s_
+        return T
+
+    def compose(self, S: "Sim2") -> "Sim2":
+        """Composition with another Sim2."""
+        return Sim2(self.R_ * S.R_, ((1.0 / S.s_) * self.t_) + self.R_ @ S.t_, self.s_ * S.s_)
+
+    def inverse(self) -> "Sim2":
+        """Return the inverse."""
+        Rt = self.R_.T
+        sRt = -Rt @ (self.s_ * self.t_)
+        return Sim2(Rt, sRt, 1.0 / self.s_)
+
+    def transform_from(self, point_cloud: np.ndarray) -> np.ndarray:
```

Comment:
Should `point_cloud` be a more generic argument name?

Reply:
it could be, e.g. `points`. But currently I've just kept it consistent with the function signature for SE(2) and SE(3)